release/release_test.go

```diff
@@ -120,32 +121,34 @@ func (s *ReleaseTestSuite) TestReleaseInfo(c *C) {
 }
 
 func (s *ReleaseTestSuite) TestForceDevMode(c *C) {
-	// Restore real OS info at the end of this function.
-	defer release.MockReleaseInfo(&release.OS{})()
-	distros := []struct {
-		id        string
-		idVersion string
-		devmode   bool
+	versionSignaturePath := filepath.Join(c.MkDir(), "version_signature")
+	restorer := release.MockVersionSignature(versionSignaturePath)
+	defer restorer()
+	apparmorPath := filepath.Join(c.MkDir(), "apparmor")
+	restorer = release.MockApparmorSysPath(apparmorPath)
+	defer restorer()
+
+	for _, t := range []struct {
+		sig string
+		aa  bool
+
+		isDevmode bool
 	}{
-		// Please keep this list sorted
-		{id: "arch", devmode: true},
-		{id: "debian", devmode: true},
-		{id: "elementary", devmode: true},
-		{id: "elementary", idVersion: "0.4", devmode: false},
-		{id: "fedora", devmode: true},
-		{id: "gentoo", devmode: true},
-		{id: "neon", devmode: false},
-		{id: "opensuse", devmode: true},
-		{id: "rhel", devmode: true},
-		{id: "ubuntu", devmode: false},
-		{id: "ubuntu-core", devmode: false},
-		{id: "linuxmint", devmode: true},
-		{id: "linuxmint", idVersion: "18.1", devmode: false},
-	}
-	for _, distro := range distros {
-		rel := &release.OS{ID: distro.id, VersionID: distro.idVersion}
-		c.Logf("checking distribution %#v", rel)
-		release.MockReleaseInfo(rel)
-		c.Assert(release.ReleaseInfo.ForceDevMode(), Equals, distro.devmode)
+		{sig: "Ubuntu 4.8.0-39.42-generic 4.8.17", aa: true, isDevmode: false},
+		{sig: "Ubuntu 4.8.0-39.42-generic 4.8.17", aa: false, isDevmode: true},
+		{sig: "Debian 4.8.0-39.42-generic 4.8.17", aa: true, isDevmode: true},
```

Comment:
I don't think Debian has `version_signature`, most distributions I checked did not. Perhaps add a test case where the string is "" and this means the file is not created.